django/core/management/commands/listurls.py

```diff
@@ -0,0 +1,277 @@
+            dest="prefixes",
+            help="Only list URLs with these prefixes.",
+            nargs="*",
+        )
+
+        # Format
+        parser.add_argument(
+            "--format",
+            "-f",
+            choices=FORMATS,
+            default="aligned",
+            dest="format",
+            help="Formatting style of the output",
+        )
+
+    def handle(self, *args, **options):
+        prefixes = options["prefixes"]
+        url_patterns = get_url_patterns(prefixes=prefixes)
+
+        if not url_patterns:
+            raise CommandError("There are no URL patterns that match given prefixes")
+
+        unsorted = options["unsorted"]
+        no_color = options["no_color"]
+        format = options["format"]
+
+        # Apply sorting
+        if not unsorted:
+            url_patterns.sort()
+
+        # Apply colors
+        self.is_color_enabled = (
+            color.supports_color()
+            and (not no_color)
+            and (format not in COLORLESS_FORMATS)
+        )
+
+        if self.is_color_enabled:
+            url_patterns = self.apply_color(url_patterns=url_patterns)
+
+        # Apply formatting
+        url_patterns = self.apply_format(
+            url_patterns=url_patterns,
+            format=format,
+        )
+
+        return url_patterns
+
+    def apply_color(self, url_patterns):
+        colored_url_patterns = []
+
+        for url_pattern in url_patterns:
+            # Route
+            route = url_pattern[0]
+            route = self.style.ROUTE(route)
+
+            # View
+            view = url_pattern[1]
+            module_path, module_name = view.rsplit(".", 1)
+            module_name = self.style.VIEW_NAME(module_name)
+            view = f"{module_path}.{module_name}"
+
+            # Name
+            name = url_pattern[2]
+            if name:
+                namespace, name = name.rsplit(":", 1) if ":" in name else ("", name)
+                name = self.style.NAME(name)
+                name = f"{namespace}:{name}" if namespace else name
+
+            # Append to the list
+            colored_url_patterns.append((route, view, name))
+
+        return colored_url_patterns
+
+    def apply_format(self, url_patterns, format):
+        format_method_name = f"format_{format.replace('-', '_')}"
+        format_method = getattr(self, format_method_name)
+
+        return format_method(url_patterns)
+
+    def format_table(self, url_patterns):
+        formatted_str = StringIO()
+
+        widths = []
+        margin = 2
+        for columns in zip(*url_patterns, strict=False):
+            widths.append(len(max(columns, key=len)) + margin)
+
+        # Headers
+        headers = ["Route", "View", "Name"]
+
+        if self.is_color_enabled:
+            headers = [self.style.HEADER(header) for header in headers]
+
+        header_parts = []
+        for width, header in zip(widths, headers, strict=False):
+            header_parts.append(header.ljust(width))
```

Comment:
Oh this is also comp potential 🕵️‍♂️

Reply:
The equivalent of this code from django-extensions was using comprehension, I chose this version for better readability.

Reply:
The comprehension version can perform better, and at the same time be readable as it's a common Python pattern.